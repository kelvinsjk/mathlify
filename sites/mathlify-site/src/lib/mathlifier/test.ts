import { mathlify } from './';

const str = mathlify`This is inline math: ${'x+2'}xyz${'k'}
and this is text.

This is display math: $${'x+2'}xyz
abcd
${'k'}

This is aligned:
$${'align'}
  x+2 &= ${'k'} \\\\
  y &= ${'z'}`;

console.log(str);
