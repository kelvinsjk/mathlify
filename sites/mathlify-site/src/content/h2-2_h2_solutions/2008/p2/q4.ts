import { Answer } from '$lib/classes/answer';
import { mathlifierDj as mathlifier } from 'mathlifier';
import { Polynomial, sum } from 'mathlify';
import { quadraticInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { inverseRelationshipSolver } from '$content/h2-1_h2_learn/01_functions/02_inverse/05_relationship/02_practice-1';
import svg from '$static/images/h2/fns/2008p2q4a.svg?raw';
import svg2 from '$static/images/h2/fns/2008p2q4c.svg?raw';

export const answer = new Answer();

// ! Complete
// (x-4)^2 + 1

const [a, b, x] = [-4, 1, 4];
const state: Parameters<typeof quadraticInverse>[0] &
	Parameters<typeof inverseRelationshipSolver>[0] = {
	fnType: 'quadratic',
	a,
	b,
	c: 0,
	restriction: {
		type: 'right',
		x,
		inclusive: false
	},
	definition: false,
	unknownConstants: false
};
const f = sum([new Polynomial([1, a]), '^', 2], b);

// a
{
	const soln = `\`\`\` =html\n${svg}\n\`\`\`\n\n`;
	const ans = mathlifier`[Sketch](#soln-1).`;
	answer.addPart(ans, soln);
}
// b
{
	const { ans, soln } = quadraticInverse(state, f);
	answer.addPart(ans, soln);
}
// c
{
	const soln = `\`\`\` =html\n${svg2}\n\`\`\`\n\n`;
	const ans = mathlifier`[Sketch](#soln-3).`;
	answer.addPart(ans, soln);
}
// d
{
	const { ans, soln } = inverseRelationshipSolver(state, f, { lineQED: true });
	//const soln = '';
	const ansLine = mathlifier`Line ${'y=x'}.`;
	answer.addPart(ansLine + '\n\n' + ans, soln);
}
