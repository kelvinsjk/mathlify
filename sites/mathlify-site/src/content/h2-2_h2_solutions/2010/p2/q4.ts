import { Answer } from '$lib/classes/answer';
import { QED } from '$lib/typesetting/utils';
import { Expression, Polynomial } from 'mathlify';
import { mathlifierDj as mathlifier } from 'mathlifier';

import { generateFn } from '$content/h2-1_h2_learn/01_functions/01_concepts/02_domain-and-range/02_practice';
import { compositeFormula } from '$content/h2-1_h2_learn/01_functions/03_composite/02_formula/02_practice';
import svg from '$static/images/h2/fns/2010p2q4a.svg?raw';
import svg2 from '$static/images/h2/fns/2010p2q4d.svg?raw';
import { Interval, intervalBuilder } from '$content/h2-1_h2_learn/01_functions/_intervals';

export const answer = new Answer();

// TODO: answer (iv) using inequalities
// f(x) = 1/(x^2-1)
// g(x) = 1/(x-3)

const [a, b] = [1, 1];
const fState: Parameters<typeof generateFn>[0] = {
	fnType: 'special',
	a,
	b,
	c: 1,
	restriction: false,
	unknownConstants: false
	//definition: false
};
const f = generateFn(fState)[1];
const g = new Polynomial([1, -3]).reciprocal();

// a
{
	const soln = svg;
	const ans = mathlifier`[Sketch](#soln-1).`;
	answer.addPart(ans, soln);
}
// b
{
	const ans = mathlifier`Least value of ${'{k=0.}'}`;
	const soln = mathlifier`Least value of ${{}} {k = 0 ${QED}}

This is because, when ${'{k=0,}'}
the domain of ${'f'}
is ${{}} [0, 1) \\cup (1, \\infty).
\\
Under this domain, all horizontal lines ${{}}{y=k, k \\in \\mathbb{R},}
cuts the graph of ${'{y=f(x)}'}
at most once so ${'f'}
is a one-to-one function and the function ${'f^{-1}'}
exists.`;
	answer.addPart(ans, soln);
}
// c
let fg: Expression;
{
	const composite = compositeFormula([f, [], g, []], true, false, {
		noDomain: true,
		factorizeDenominator: true
	});
	const { ans, soln } = composite;
	({ exp: fg } = composite);
	answer.addPart(ans, soln);
}
// d
{
	const soln =
		mathlifier`$${fg} > 0` + svg2 + mathlifier`$${{}} 2 < x < 3 \\; \\text{ or } \\; 3 < x < 4`;
	const ans = mathlifier`${{}}{2 < x < 3}
or ${{}} {3 < x < 4.}`;
	answer.addPart(ans, soln);
}
// e
{
	const Rfg = [intervalBuilder('left', -1, false), intervalBuilder('right', 0, false)].join(
		' \\cup '
	);
	const ans = mathlifier`${'R_{fg}'} = ${Rfg}.`;
	const soln = mathlifier`$${'align*'}
	D_{g} &= ${[intervalBuilder('left', 2, false), new Interval({ left: 2, right: 3 }), new Interval({ left: 3, right: 4 }), intervalBuilder('right', 4, false)].join(' \\cup ')} \\\\
	&\\downarrow g \\\\
	R_{g} &= ${[intervalBuilder('left', -1, false), new Interval({ left: -1, right: 0 }), new Interval({ left: 0, right: 1 }), intervalBuilder('right', 1, false)].join(' \\cup ')} \\\\
	&\\downarrow f \\\\
	R_{fg} &= ${Rfg}
`;
	answer.addPart(ans, soln);
}
