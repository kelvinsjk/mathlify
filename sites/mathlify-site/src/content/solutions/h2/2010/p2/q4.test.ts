import { expect, it } from 'vitest';
import { answer } from './q4.soln';
import { mathlify } from '$lib/mathlifier';

const ans = answer.answer;
const actual: typeof ans = {
	parts: [
		{ body: mathlify`[Sketch](#soln-1).` },
		{ body: mathlify`Least value of ${{}} {k=0.}` },
		{
			body: mathlify`${{}} fg(x) = \\frac{\\left(x-3\\right)^2}{\\left(4-x\\right)\\left(x-2\\right)}.`,
		},
		{
			body: mathlify`${{}} {2 < x < 3}
or ${{}} {3 < x < 4.}`,
		},
		{
			body: mathlify`${{}} R_{fg} = \\left(-\\infty, -1\\right) \\cup \\left( 0, \\infty \\right).`,
		},
	],
};
it('2010p2q4', () => {
	expect(answer).toMatchSnapshot();
	expect(ans).toEqual(actual);
});
