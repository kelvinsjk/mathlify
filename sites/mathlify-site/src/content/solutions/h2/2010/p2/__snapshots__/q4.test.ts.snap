// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2010p2q4 1`] = `
Answer {
  "answer": {
    "parts": [
      {
        "body": "<p><a href="#soln-1">Sketch</a>.</p>
",
      },
      {
        "body": "<p>Least value of <math><mrow><mi>k</mi><mo>=</mo><mn>0</mn><mi>.</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math><mrow><mi>f</mi><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mfrac><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>4</mn><mo>−</mo><mi>x</mi><mo fence="true" form="postfix">)</mo></mrow><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac><mi>.</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math><mrow><mn>2</mn><mo>&lt;</mo><mi>x</mi><mo>&lt;</mo><mn>3</mn></mrow></math>
or <math><mrow><mn>3</mn><mo>&lt;</mo><mi>x</mi><mo>&lt;</mo><mn>4</mn><mi>.</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math><mrow><msub><mi>R</mi><mrow><mi>f</mi><mi>g</mi></mrow></msub><mo>=</mo></mrow><mrow><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo></mrow><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mi>.</mi></mrow></math></p>
",
      },
    ],
  },
  "solution": {
    "parts": [
      {
        "body": "<svg
	xmlns="http://www.w3.org/2000/svg"
	class="graph svelte-1vox4to"
	viewBox="-55 -60 110.00000000000001 110.00000000000001"
>
	<style>
		foreignObject {
			color: black;
			font-size: 7px;
			overflow: visible;
			white-space: nowrap;
			display: grid;
		}
		foreignObject math {
			flex-wrap: nowrap;
			transform: translate(-50%, -50%);
		}
		foreignObject div.sm {
			font-size: 5px;
		}
		foreignObject div.xs {
			font-size: 3px;
		}
	</style>
	<defs>
		<marker id="closed" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle cx="1" cy="1" r="1" fill="black"></circle>
		</marker>
		<marker id="open" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle
				cx="1"
				cy="1"
				r="0.75"
				fill="white"
				markerUnits="strokeWidth"
				style="stroke: black; stroke-width: 0.5"
			></circle>
		</marker>
		<marker id="arrow" markerWidth="5" markerHeight="5" refX="0" refY="1" orient="auto">
			<path d="M 0 0 L 2 1 L 0 2 z" fill="black"></path>
		</marker>
	</defs>
	<line
		marker-end="url(#arrow)"
		x1="-50"
		x2="50"
		y1="0"
		y2="0"
		style="stroke: black; stroke-width: 2"
	></line>
	<line
		marker-end="url(#arrow)"
		x1="0"
		x2="0"
		y1="50"
		y2="-50"
		style="stroke: black; stroke-width: 2"
	></line>
	<foreignObject width="1" height="1" class="svelte-1vox4to" x="50" y="5"
		><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math
		><!----></foreignObject
	><foreignObject width="1" height="1" class="svelte-1vox4to" x="-5" y="-52"
		><math xmlns="http://www.w3.org/1998/Math/MathML"><mi>y</mi></math
		><!----></foreignObject
	><foreignObject width="1" height="1" class="svelte-1vox4to" x="-5" y="5"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>0</mn></math
			><!---->
		</div></foreignObject
	><!----><!----><!---->
	<line stroke-dasharray="4" x1="-50" x2="50" y1="0" y2="0" style="stroke: black"></line>
	<foreignObject class="graph-label svelte-1vox4to" width="1" height="1" x="-50" y="5"
		><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow><mn>0</mn></mrow></math
		><!----></foreignObject
	><!---->
	<line stroke-dasharray="4" x1="-10" x2="-10" y1="50" y2="-50" style="stroke: green"></line>
	<foreignObject class="graph-label svelte-1vox4to" width="1" height="1" x="-22" y="45"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math xmlns="http://www.w3.org/1998/Math/MathML"
				><mrow><mi>x</mi><mo>=</mo></mrow
				><mrow><mo form="prefix" stretchy="false">−</mo></mrow
				><mrow><mn>1</mn></mrow></math
			><!---->
		</div></foreignObject
	><!---->
	<line stroke-dasharray="4" x1="10" x2="10" y1="50" y2="-50" style="stroke: green"></line>
	<foreignObject class="graph-label svelte-1vox4to" width="1" height="1" x="22" y="45"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1vox4to">
			<math xmlns="http://www.w3.org/1998/Math/MathML"
				><mrow><mi>x</mi><mo>=</mo></mrow
				><mrow><mn>1</mn></mrow></math
			><!---->
		</div></foreignObject
	><!----><!---->
	<path
		d="M-49.000,-0.43459 -48.224,-0.44933 -47.448,-0.46483 -46.672,-0.48117 -45.896,-0.49840 -45.120,-0.51659 -44.344,-0.53580 -43.568,-0.55613 -42.792,-0.57766 -42.015,-0.60049 -41.239,-0.62473 -40.463,-0.65050 -39.687,-0.67793 -38.911,-0.70717 -38.135,-0.73839 -37.359,-0.77178 -36.583,-0.80755 -35.807,-0.84592 -35.031,-0.88718 -34.255,-0.93162 -33.479,-0.97959 -32.703,-1.0315 -31.927,-1.0878 -31.151,-1.1489 -30.375,-1.2156 -29.599,-1.2885 -28.822,-1.3685 -28.046,-1.4565 -27.270,-1.5536 -26.494,-1.6613 -25.718,-1.7812 -24.942,-1.9153 -24.166,-2.0661 -23.390,-2.2367 -22.614,-2.4308 -21.838,-2.6533 -21.062,-2.9103 -20.286,-3.2101 -19.510,-3.5634 -18.734,-3.9848 -17.958,-4.4949 -17.182,-5.1228 -16.406,-5.9122 -15.629,-6.9310 -14.853,-8.2903 -14.077,-10.186 -13.301,-13.000 -12.525,-17.581 -11.749,-26.286 -10.973,-48.998 "
		style="stroke: blue; fill: none"
		stroke-dasharray="0"
	></path>
	<!---->
	<path
		d="M-8.9214,48.999 -8.3061,32.250 -7.6909,24.479 -7.0756,20.026 -6.4603,17.163 -5.8451,15.189 -5.2298,13.765 -4.6145,12.705 -3.9992,11.904 -3.3840,11.293 -2.7687,10.830 -2.1534,10.486 -1.5382,10.242 -0.92290,10.086 -0.30763,10.009 0.30763,10.009 0.92290,10.086 1.5382,10.242 2.1534,10.486 2.7687,10.830 3.3840,11.293 3.9992,11.904 4.6145,12.705 5.2298,13.765 5.8451,15.189 6.4603,17.163 7.0756,20.026 7.6909,24.479 8.3061,32.250 8.9214,48.999 "
		style="stroke: blue; fill: none"
		stroke-dasharray="0"
	></path>
	<!---->
	<path
		d="M10.973,-48.998 11.749,-26.286 12.525,-17.581 13.301,-13.000 14.077,-10.186 14.853,-8.2903 15.629,-6.9310 16.406,-5.9122 17.182,-5.1228 17.958,-4.4949 18.734,-3.9848 19.510,-3.5634 20.286,-3.2101 21.062,-2.9103 21.838,-2.6533 22.614,-2.4308 23.390,-2.2367 24.166,-2.0661 24.942,-1.9153 25.718,-1.7812 26.494,-1.6613 27.270,-1.5536 28.046,-1.4565 28.822,-1.3685 29.599,-1.2885 30.375,-1.2156 31.151,-1.1489 31.927,-1.0878 32.703,-1.0315 33.479,-0.97959 34.255,-0.93162 35.031,-0.88718 35.807,-0.84592 36.583,-0.80755 37.359,-0.77178 38.135,-0.73839 38.911,-0.70717 39.687,-0.67793 40.463,-0.65050 41.239,-0.62473 42.015,-0.60049 42.792,-0.57766 43.568,-0.55613 44.344,-0.53580 45.120,-0.51659 45.896,-0.49840 46.672,-0.48117 47.448,-0.46483 48.224,-0.44933 49.000,-0.43459 "
		style="stroke: blue; fill: none"
		stroke-dasharray="0"
	></path>
	<foreignObject
		class="graph-label svelte-1vox4to"
		width="1"
		height="1"
		x="13.973100000000002"
		y="-23.998174480403208"
		><math xmlns="http://www.w3.org/1998/Math/MathML"
			><mrow><mi>y</mi><mo>=</mo></mrow
			><mrow
				><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi
				><mo form="postfix" stretchy="false">)</mo></mrow
			></math
		><!----></foreignObject
	><!----><!----><!---->
	<line style="stroke: black" x1="-3" x2="3" y1="10" y2="10"></line>
	<foreignObject width="1" height="1" class="svelte-1vox4to" x="-12" y="10"
		><div
			class="label-left svelte-1vox4to"
			style="display: grid"
			xmlns="http://www.w3.org/1999/xhtml"
		>
			<math xmlns="http://www.w3.org/1998/Math/MathML"
				><mrow
					><mo form="prefix" stretchy="false">(</mo><mn>0</mn><mo separator="true">,</mo
					><mo form="prefix" stretchy="false">−</mo><mn>1</mn
					><mo form="postfix" stretchy="false">)</mo></mrow
				></math
			><!---->
		</div></foreignObject
	><!----><!---->
</svg>
",
      },
      {
        "body": "<p>Least value of <math><mrow><mi>k</mi><mo>=</mo><mn>0</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
<p>This is because, when <math><mrow><mi>k</mi><mo>=</mo><mn>0</mn><mo separator="true" lspace="0em" rspace="0em">,</mo></mrow></math>
the domain of <math><mi>f</mi></math>
is <math><mrow><mo form="prefix" stretchy="false">[</mo><mn>0</mn><mo separator="true">,</mo><mn>1</mn><mo form="postfix" stretchy="false">)</mo><mo>∪</mo></mrow><mrow><mo form="prefix" stretchy="false">(</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo form="postfix" stretchy="false">)</mo><mi>.</mi></mrow></math>
<br>
Under this domain, all horizontal lines <math><mrow><mi>y</mi><mo>=</mo><mi>k</mi><mo separator="true">,</mo><mi>k</mi><mo>∈</mo><mi>ℝ</mi><mo separator="true" lspace="0em" rspace="0em">,</mo></mrow></math>
cuts the graph of <math><mrow><mi>y</mi><mo>=</mo><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false" lspace="0em" rspace="0em">)</mo></mrow></math>
at most once so <math><mi>f</mi></math>
is a one-to-one function and the function <math><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></math>
exists.<math><mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>f</mi><mi>g</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>f</mi><mrow><mo fence="true" form="prefix">(</mo><mfrac><mn>1</mn><mrow><mi>x</mi><mo>−</mo><mn>3</mn></mrow></mfrac><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mn>1</mn><mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mfrac><mn>1</mn><mrow><mi>x</mi><mo>−</mo><mn>3</mn></mrow></mfrac><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mn>1</mn><mrow><mfrac><mn>1</mn><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac><mo>−</mo><mn>1</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mn>1</mn><mfrac><mrow><mo lspace="0em" rspace="0em">−</mo><mn>8</mn><mo>−</mo><msup><mi>x</mi><mn>2</mn></msup><mo>+</mo><mn>6</mn><mi>x</mi></mrow><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mrow><mo lspace="0em" rspace="0em">−</mo><mn>8</mn><mo>−</mo><msup><mi>x</mi><mn>2</mn></msup><mo>+</mo><mn>6</mn><mi>x</mi></mrow></mfrac></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>4</mn><mo>−</mo><mi>x</mi><mo fence="true" form="postfix">)</mo></mrow><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow></mtd></mtr></mtable></math></p>
",
      },
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mrow><mfrac><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>4</mn><mo>−</mo><mi>x</mi><mo fence="true" form="postfix">)</mo></mrow><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow></mrow></mfrac><mo>&gt;</mo><mn>0</mn></mrow></math></p>
<svg
	xmlns="http://www.w3.org/2000/svg"
	class="critical-value-number-line svelte-1ew0o84"
	viewBox="-55 -15 120 36"
>
	<style>
		foreignObject {
			color: black;
			font-size: 7px;
			overflow: visible;
			white-space: nowrap;
			display: grid;
		}
		foreignObject math {
			flex-wrap: nowrap;
			align-items: baseline;
			white-space: nowrap;
			transform: translate(-50%, -50%);
		}
	</style>
	<defs>
		<marker id="closed" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle cx="1" cy="1" r="1" fill="black"></circle>
		</marker>
		<marker id="open" markerWidth="5" markerHeight="5" refX="1" refY="1">
			<circle
				cx="1"
				cy="1"
				r="0.75"
				fill="white"
				markerUnits="strokeWidth"
				style="stroke: black; stroke-width: 0.5"
			></circle>
		</marker>
		<marker id="arrow" markerWidth="5" markerHeight="5" refX="0" refY="1" orient="auto">
			<path d="M 0 0 L 2 1 L 0 2 z" fill="black"></path>
		</marker>
	</defs>
	<line
		marker-end="url(#arrow)"
		x1="-50"
		x2="55"
		y1="0"
		y2="0"
		style="stroke: black; stroke-width: 2"
	></line>
	<foreignObject y="5" width="1" height="1" class="svelte-1ew0o84" x="59"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mi>x</mi></math
			><!---->
		</div></foreignObject
	>
	<line x1="-25" x2="-25" y1="-4" y2="4" style="stroke: black; stroke-width: 1"></line>
	<foreignObject width="1" height="1" class="svelte-1ew0o84" x="-25" y="10"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>2</mn></math
			><!---->
		</div></foreignObject
	><foreignObject width="1" height="1" class="svelte-1ew0o84" x="-37.5" y="-7"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mo lspace="0em" rspace="0em">−</mo></math
			><!---->
		</div></foreignObject
	><!---->
	<line x1="0" x2="0" y1="-4" y2="4" style="stroke: black; stroke-width: 1"></line>
	<foreignObject width="1" height="1" class="svelte-1ew0o84" x="0" y="10"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>3</mn></math
			><!---->
		</div></foreignObject
	><foreignObject width="1" height="1" class="svelte-1ew0o84" x="-12.5" y="-7"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mo lspace="0em" rspace="0em">+</mo></math
			><!---->
		</div></foreignObject
	><!---->
	<line x1="25" x2="25" y1="-4" y2="4" style="stroke: black; stroke-width: 1"></line>
	<foreignObject width="1" height="1" class="svelte-1ew0o84" x="25" y="10"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mn>4</mn></math
			><!---->
		</div></foreignObject
	><foreignObject width="1" height="1" class="svelte-1ew0o84" x="12.5" y="-7"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mo lspace="0em" rspace="0em">+</mo></math
			><!---->
		</div></foreignObject
	><foreignObject width="1" height="1" class="svelte-1ew0o84" x="37.5" y="-7"
		><div style="display: grid" xmlns="http://www.w3.org/1999/xhtml" class="svelte-1ew0o84">
			<math xmlns="http://www.w3.org/1998/Math/MathML"><mo lspace="0em" rspace="0em">−</mo></math
			><!---->
		</div></foreignObject
	><!----><!---->
</svg>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mn>2</mn><mo>&lt;</mo><mi>x</mi><mo>&lt;</mo><mn>3</mn><mspace width="0.2778em"></mspace><mtext> or </mtext><mspace width="0.2778em"></mspace><mn>3</mn><mo>&lt;</mo><mi>x</mi><mo>&lt;</mo><mn>4</mn><mspace width="0.2778em"></mspace><mi>■</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>D</mi><mi>g</mi></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mn>2</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>2</mn><mo separator="true">,</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>3</mn><mo separator="true">,</mo><mn>4</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>4</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo stretchy="false">↓</mo><mi>g</mi></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>R</mi><mi>g</mi></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mo separator="true">,</mo><mn>0</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo separator="true">,</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>1</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo stretchy="false">↓</mo><mi>f</mi></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>R</mi><mrow><mi>f</mi><mi>g</mi></mrow></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mo form="prefix" stretchy="false">−</mo><mn>1</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mspace width="0.2778em"></mspace><mi>■</mi></mrow></mtd></mtr></mtable></math></p>
",
      },
    ],
  },
}
`;
